src/monitoring_tests/fee_monitoring/fee_monitoring.py

```diff
@@ -0,0 +1,239 @@
+"""
+Fee Monitoring
+"""
+
+import json
+from typing import Tuple
+import requests
+from src.monitoring_tests.template_test import TemplateTest
+from src.constants import SUCCESS_CODE, header
+from src.helper_functions import DecodedSettlement
+
+
+class FeeMonitoring:
+    """
+    Class for fee monitoring.
+    """
+
+    def get_order_execution(self, order, tx_hash) -> Tuple[int, int, int]:
+        """
+        Given an order and a transaction hash, compute buy_amount, sell_amount, and fee_amount
+        of the trade.
+        """
+        order_uid = order["uid"]
+        prod_endpoint_url = (
+            "https://api.cow.fi/mainnet/api/v1/trades?orderUid=" + order_uid
+        )
+        trades_response = requests.get(
+            prod_endpoint_url,
+            headers=header,
+            timeout=30,
+        )
+        trades = json.loads(trades_response.text)
+        for trade in trades:
+            if trade["txHash"] == tx_hash:
+                trade_0 = trade
+                break
+
+        fee_amount = TemplateTest.get_fee(order, tx_hash)
+        sell_amount = int(trade_0["sellAmount"]) - fee_amount
+        buy_amount = int(trade_0["buyAmount"])
+
+        return buy_amount, sell_amount, fee_amount
+
+    def get_quote(self, decoded_settlement, i) -> Tuple[int, int, int]:
+        """
+        Given a trade, compute buy_amount, sell_amount, and fee_amount of the trade
+        as proposed by our quoting infrastructure.
+        """
+        trade = decoded_settlement.trades[i]
+
+        if str(f"{trade['flags']:08b}")[-1] == "0":
+            kind = "sell"
+        else:
+            kind = "buy"
+
+        request_dict = {
+            "sellToken": decoded_settlement.tokens[trade["sellTokenIndex"]],
+            "buyToken": decoded_settlement.tokens[trade["buyTokenIndex"]],
+            "receiver": trade["receiver"],
+            "appData": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "partiallyFillable": False,
+            "sellTokenBalance": "erc20",
+            "buyTokenBalance": "erc20",
+            "from": trade["receiver"],
+            "priceQuality": "optimal",
+            "signingScheme": "eip712",
+            "onchainOrder": False,
+            "kind": kind,
+            "sellAmountBeforeFee": str(trade["executedAmount"]),
+        }
+
+        try:
+            prod_endpoint_url = "https://api.cow.fi/mainnet/api/v1/quote"
+            quote_response = requests.post(
+                prod_endpoint_url,
+                headers=header,
+                json=request_dict,
+                timeout=30,
+            )
+            if quote_response.status_code != SUCCESS_CODE:
+                barn_endpoint_url = "https://barn.api.cow.fi/mainnet/api/v1/quote"
```

Comment:
for the quotes, I would actually only use the prod api; if it is down then we have a problem anyways...